src/algorithms/hilbert_curve.rs

```diff
@@ -210,36 +204,33 @@ fn segment_to_segment(min: f64, max: f64, order: usize) -> impl Fn(f64) -> u64 {
     }
 }
 
-/// Returns a function that maps 2D points to their hilbert curve index.
+/// Returns a function that maps points to their curve index.
 ///
 /// Panics if `points` is empty.
-fn index_fn_2d(points: &[Point2D], order: usize) -> impl Fn(&Point2D) -> u64 {
-    let mbr = OrientedBoundingBox::from_points(points).unwrap();
-    let aabb = mbr.aabb();
-    let x_mapping = segment_to_segment(aabb.p_min.x, aabb.p_max.x, order);
-    let y_mapping = segment_to_segment(aabb.p_min.y, aabb.p_max.y, order);
-    move |p| {
-        let p = mbr.obb_to_aabb(p);
-        encode_2d(x_mapping(p.x), y_mapping(p.y), order)
-    }
-}
-
-/// Returns a function that maps 3D points to their hilbert curve index.
-///
-/// Panics if `points` is empty.
-fn index_fn_3d(points: &[Point3D], order: usize) -> impl Fn(&Point3D) -> u64 {
+fn index_fn<const D: usize>(
+    points: &[PointND<D>],
+    order: usize,
+    curve: impl Fn([u64; D]) -> u64,
+) -> impl Fn(&PointND<D>) -> u64
+where
+    Const<D>: DimSub<Const<1>>,
+    DefaultAllocator: Allocator<f64, Const<D>, Const<D>, Buffer = ArrayStorage<f64, D, D>>
+        + Allocator<f64, DimDiff<Const<D>, Const<1>>>,
+{
```

Comment:
This is due to `nalgebra` requirements ?

Reply:
Yes actually it stems from [`nalgebra::Matrix::symmetric_eigen`][0] called here:

https://github.com/LIHPC-Computational-Geometry/coupe/blob/1e9ecb63844b8a170535a0d81282b73f37556ee6/src/geometry.rs#L286

I guess until rust allows bounds like `where D > 0` we're stuck with nalgebra's workarounds. Or we could use `BoundingBox` instead of `OrientedBoundingBox`, though the partition quality might drop (needs testing).

[0]: https://docs.rs/nalgebra/latest/nalgebra/base/struct.Matrix.html#method.symmetric_eigen